File: UNIDAD 1/01_SOA_CON_BDD/00. SOAP_JAVA_EUREKABANK_GR10/02. CLICON/EUREKABANK_CLICON_GR10/src/ec/edu/monster/controlador/EurekaController.java
package ec.edu.monster.controlador;

import ec.edu.monster.servicio.EurekaService;
import ec.edu.monster.ws.Movimiento;
import java.util.List;

/**
 * Controller to coordinate between console input and service layer.
 *
 * @author MATIAS
 */
public class EurekaController {

    private final EurekaService service;

    public EurekaController() {
        this.service = new EurekaService();
    }

    public List<Movimiento> consultarMovimientos(String cuenta) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        List<Movimiento> movimientos = service.traerMovimientos(cuenta);
        if (movimientos.isEmpty()) {
            throw new Exception("No se encontraron movimientos para la cuenta.");
        }
        return movimientos;
    }

    public boolean registrarDeposito(String cuenta, double importe, String codEmp) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        if (importe <= 0) {
            throw new Exception("El importe debe ser mayor que 0.");
        }
        if (codEmp == null || codEmp.trim().isEmpty()) {
            throw new Exception("El código de empleado es requerido.");
        }
        return service.registrarDeposito(cuenta, importe, codEmp) == 1;
    }

    public boolean registrarRetiro(String cuenta, double importe, String codEmp) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        if (importe <= 0) {
            throw new Exception("El importe debe ser mayor que 0.");
        }
        if (codEmp == null || codEmp.trim().isEmpty()) {
            throw new Exception("El código de empleado es requerido.");
        }
        double costo = service.obtenerCostoMovimiento(cuenta);
        if (costo == -1.0) {
            throw new Exception("Error al obtener el costo del movimiento.");
        }
        return service.registrarRetiro(cuenta, importe, codEmp) == 1;
    }

    public boolean realizarTransferencia(String cuentaOrigen, String cuentaDestino, double importe, String codEmp) throws Exception {
        if (cuentaOrigen == null || cuentaOrigen.trim().isEmpty() || cuentaDestino == null || cuentaDestino.trim().isEmpty()) {
            throw new Exception("Los números de cuenta son requeridos.");
        }
        if (importe <= 0) {
            throw new Exception("El importe debe ser mayor que 0.");
        }
        if (codEmp == null || codEmp.trim().isEmpty()) {
            throw new Exception("El código de empleado es requerido.");
        }
        double costo = service.obtenerCostoMovimiento(cuentaOrigen);
        if (costo == -1.0) {
            throw new Exception("Error al obtener el costo del movimiento.");
        }
        return service.realizarTransferencia(cuentaOrigen, cuentaDestino, importe, codEmp) == 1;
    }

    public double verificarSaldo(String cuenta) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        double saldo = service.verificarSaldo(cuenta);
        if (saldo == -1.0) {
            throw new Exception("Error al verificar el saldo.");
        }
        return saldo;
    }

    public double obtenerCostoMovimiento(String cuenta) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        double costo = service.obtenerCostoMovimiento(cuenta);
        if (costo == -1.0) {
            throw new Exception("Error al obtener el costo del movimiento.");
        }
        return costo;
    }

    /**
     * Inicia sesión para un usuario.
     *
     * @param usuario Nombre de usuario (vch_emplusuario).
     * @param clave Clave del usuario (vch_emplclave).
     * @return true si la autenticación es exitosa, false si falla.
     * @throws Exception si ocurre un error en el proceso.
     */
    public boolean iniciarSesion(String usuario, String clave) throws Exception {
        if (usuario == null || usuario.trim().isEmpty() || clave == null || clave.trim().isEmpty()) {
            throw new Exception("Usuario y contraseña son requeridos.");
        }
        ec.edu.monster.modelo.Usuario user = service.iniciarSesion(usuario, clave);
        return user != null;
    }
}

File: UNIDAD 1/01_SOA_CON_BDD/001. SOAP_DOTNET_EUREKABANK_GR10/02. CLICON/EurekaBankConsoleClient/EurekaBankConsoleClient/Controlador/EurekaController.cs
using EurekaBankConsoleClient.Servicio;
using EurekaServiceReference;
using System;
using System.Collections.Generic;

namespace EurekaBankConsoleClient.Controlador
{
    public class EurekaController
    {
        private readonly EurekaService service;

        public EurekaController()
        {
            this.service = new EurekaService();
        }

        public List<Movimiento> ConsultarMovimientos(string cuenta)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            List<Movimiento> movimientos = service.TraerMovimientos(cuenta);
            if (movimientos.Count == 0)
            {
                throw new Exception("No se encontraron movimientos para la cuenta.");
            }
            return movimientos;
        }

        public bool RegistrarDeposito(string cuenta, double importe, string codEmp)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            if (importe <= 0)
            {
                throw new Exception("El importe debe ser mayor que 0.");
            }
            if (string.IsNullOrWhiteSpace(codEmp))
            {
                throw new Exception("El código de empleado es requerido.");
            }
            return service.RegistrarDeposito(cuenta, importe, codEmp) == 1;
        }

        public bool RegistrarRetiro(string cuenta, double importe, string codEmp)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            if (importe <= 0)
            {
                throw new Exception("El importe debe ser mayor que 0.");
            }
            if (string.IsNullOrWhiteSpace(codEmp))
            {
                throw new Exception("El código de empleado es requerido.");
            }
            double costo = service.ObtenerCostoMovimiento(cuenta);
            if (costo == -1.0)
            {
                throw new Exception("Error al obtener el costo del movimiento.");
            }
            return service.RegistrarRetiro(cuenta, importe, codEmp) == 1;
        }

        public bool RealizarTransferencia(string cuentaOrigen, string cuentaDestino, double importe, string codEmp)
        {
            if (string.IsNullOrWhiteSpace(cuentaOrigen) || string.IsNullOrWhiteSpace(cuentaDestino))
            {
                throw new Exception("Los números de cuenta son requeridos.");
            }
            if (importe <= 0)
            {
                throw new Exception("El importe debe ser mayor que 0.");
            }
            if (string.IsNullOrWhiteSpace(codEmp))
            {
                throw new Exception("El código de empleado es requerido.");
            }
            double costo = service.ObtenerCostoMovimiento(cuentaOrigen);
            if (costo == -1.0)
            {
                throw new Exception("Error al obtener el costo del movimiento.");
            }
            return service.RealizarTransferencia(cuentaOrigen, cuentaDestino, importe, codEmp) == 1;
        }

        public double VerificarSaldo(string cuenta)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            double saldo = service.VerificarSaldo(cuenta);
            if (saldo == -1.0)
            {
                throw new Exception("Error al verificar el saldo.");
            }
            return saldo;
        }

        public double ObtenerCostoMovimiento(string cuenta)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            double costo = service.ObtenerCostoMovimiento(cuenta);
            if (costo == -1.0)
            {
                throw new Exception("Error al obtener el costo del movimiento.");
            }
            return costo;
        }

        public bool IniciarSesion(string usuario, string clave)
        {
            if (string.IsNullOrWhiteSpace(usuario) || string.IsNullOrWhiteSpace(clave))
            {
                throw new Exception("Usuario y contraseña son requeridos.");
            }
            var user = service.IniciarSesion(usuario, clave);
            return user != null;
        }
    }
}